Yona/Yona/AppDelegate.swift

```diff
@@ -69,6 +67,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate,URLSessionDelegate {
         return true
         
     }
+    
+    func initializeCrashlytics(){
+        var fabricAPIKey: String? = nil
+        if let filepath = Bundle.main.path(forResource: "fabric.apikey", ofType: nil) {
+            do {
+                fabricAPIKey = try String(contentsOfFile: filepath)
+            } catch {
+                print(error.localizedDescription)
```

Comment:
Shouldn't the app fail to start when this resource is not part of the app?

Reply:
That's the reason we added empty file ''' fabric.apikey'''  into the git repo. And writing key into file from environment variable just before the ``` xcodebuild archive ``` step.

Reply:
Understood, but that can be handled differently.
Just put the file inside your workspace, without adding it to Git and you're done with the file existence.

Next, make start up fail at runtime when the resource doesn't exist or cannot be read.

Reply:
File should exist and also be linked against the build target like other resource files, for xcodebuild command to copy the it into the final ipa file.
Or we need to run a ``` cp ``` shell script to do that. Not sure whether it works. need to test that.

Reply:
Yes, you need the file, but it doesn't need to be checked in on Git. Just create the file locally. And at runtime, make it fail if it wasn't built correctly, with a descriptive error message.

Reply:
Even when the file is not found, what you meant here is to continue with an error message to the user as this is not a compulsory requirement for app to run. right?

Reply:
If this fails, the app is not built the right way. To make sure that doesn't go unnoticed, the app should fail wit a clear error message.

Reply:
We can have a Test case for this. where app would not build at all if the file does not exist.